gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource.java

```diff
@@ -277,6 +301,40 @@ public Response exportApiDefinition(@Context HttpHeaders headers, @PathParam("ap
             .build();
     }
 
+    @POST
+    @Path("/_duplicate")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Permissions(
+        {
+            @Permission(value = RolePermission.API_DEFINITION, acls = RolePermissionAction.READ),
+            @Permission(value = RolePermission.ENVIRONMENT_API, acls = RolePermissionAction.CREATE),
+        }
+    )
+    public Response duplicateAPI(@PathParam("apiId") String apiId, @Valid @NotNull DuplicateApiOptions duplicateOptions) {
+        final GenericApiEntity currentEntity = getGenericApiEntityById(apiId, false);
+
+        GenericApiEntity duplicate;
+        var definitionVersion = currentEntity.getDefinitionVersion();
+        if (definitionVersion == DefinitionVersion.V4) {
+            duplicate =
+                duplicateApiService.duplicate(
+                    GraviteeContext.getExecutionContext(),
+                    (ApiEntity) currentEntity,
+                    DuplicateApiMapper.INSTANCE.mapToDuplicateOptions(duplicateOptions)
+                );
+        } else {
```

Comment:
I think duplicating V1 API should not be allowed. Do we need to return a 400 bad request in that case ?